### PR TITLE
Add section on smart nested field

### PR DESCRIPTION
**Pull request by MyGodIsHe**
This is a small recipe for improving the nested field. I spent a lot of energy to find a place where I can do this trick. I think this is a common problem for rest api.

### CODE REVIEW

File: docs/recipes.rst [hunk 1]
```diff
@@ -219,3 +219,43 @@ An example of then using this:
...
+Smart Nested Field
+======================================================
+If you want to avoid lazy loading a nested attribute, but you still need id, then use this trick.
+
+.. code-block:: python
+
+    class SmartNested(fields.Nested):
+
+        def serialize(self, attr, obj, accessor=None):
+            if attr not in obj.__dict__:
```
Comment (sloria) [inline docs/recipes.rst:233]: Why not use `hasattr`?

File: docs/recipes.rst [hunk 2]
```diff
@@ -219,3 +219,43 @@ An example of then using this:
     print author_schema.dump(author).data
 
 This is inspired by functionality from ColanderAlchemy.
+
+Smart Nested Field
+======================================================
```
Comment (sloria) [inline docs/recipes.rst]: The underline should be the same length as the title; otherwise Sphinx will raise a warning.

File: docs/recipes.rst [hunk 3]
```diff
@@ -219,3 +219,43 @@ An example of then using this:
...
+If you want to avoid lazy loading a nested attribute, but you still need id, then use this trick.
```
Comment (sloria) [inline docs/recipes.rst]: How about revising this to:

"To serialize nested attributes to primary keys unless they are already loaded, you can use this custom field."

File: docs/recipes.rst [hunk 4]
```diff
@@ -219,3 +219,43 @@ An example of then using this:
...
+
+    book = Book.query.get(1)
+    print(BookSchema().dump(book).data['author'])
+    # {'id': 1}
+
+    book = Book.query.options(joinedload('author')).get(1)
+    print(BookSchema().dump(book).data['author'])
+    # {'id': 1, 'name': 'Chuck Paluhniuk'}
+
+This method allows you to make convenient rest api, where relationship id will always be in one place.
```
Comment (sloria) [inline docs/recipes.rst]: This sentence is unnecessary; omit.

**Review by sloria — CHANGES_REQUESTED** [review]
Thanks for the contribution! This is a neat recipe. Just a few revisions, and this is will be good to merge.

File: docs/recipes.rst (hunk 1 above)
Comment (MyGodIsHe) [reply docs/recipes.rst:233]: This lazy attribute always exists. So I need to test this in a different way.

Comment (sloria) [reply docs/recipes.rst:233]: Ah ok, thanks for the clarification.

File: docs/recipes.rst (hunk 3 above)
Comment (sloria) [reply docs/recipes.rst]: Also, I believe there needs to be an empty line between the header and this text in order to be rendered correctly.

**Review by sloria — APPROVED** [review]

**Comment by sloria** [comment]
Thanks!

**Pull request merged**